test/e2e/utils/recordedit-helpers.js

```diff
@@ -169,7 +207,38 @@ exports.testPresentationAndBasicValidation = function(tableParams, isEditMode) {
                 });
             });
 
+            it('should render correct markdown in modal box.', function() {
+              var descColList = tableParams.columns.filter(function(c) {
+                if ((c.type === "markdown") && !c.isForeignKey) return true;
+              });
+              descColList.forEach(function(descCol) {
+                var markdownField = chaisePage.recordEditPage.getInputById(0, descCol.title);
+                var livePreviewLink = element(by.className('live-preview'));
+                for (i = 0; i < markdownTestParams.length; i++) {
+                  markdownField.clear();
+                  (function(input, markdownOut) {
+                    markdownField.sendKeys(input);
+                    livePreviewLink.click();
+                    browser.getAllWindowHandles().then(function(handles) {
+                      browser.switchTo().window(handles[0]).then(function() {
```

Comment:
What is the purpose of get the window handles here? You can just grab the modal that pops up because it pops up in the same `window`.

We have other tests, ones that test delete confirmation come to mind, that wait for a modal window to open and then continue testing once that has occurred.